src/hosted_db/paths.rs

```diff
@@ -21,35 +25,75 @@ pub fn database_parent_path(data_path: &str, create_path: bool) -> Result<PathBu
     Ok(fs::canonicalize(path)?)
 }
 
-pub fn database_path(
+/// Returns a path for a new database directory for storing
+/// `{entity_slug}/{database_slug}`. The format for this path is
+/// `{data_path}/{entity_slug}/{database_slug}/{time_sortable_uuid}/`.
+pub fn new_database_path(
     entity_slug: &str,
     database_slug: &str,
     data_path: &str,
-    create_database: bool,
 ) -> Result<PathBuf, AybError> {
+    let uuid = Uuid::new_v7(Timestamp::now(ContextV7::new()));
     // We place each database in its own directory because databases
     // might span multiple files (e.g, the SQLite database file as
     // well as a journal/write-ahead log).
-    let mut path: PathBuf = [data_path, DATABASES, entity_slug, database_slug]
-        .iter()
-        .collect();
-    if create_database {
-        if let Err(e) = fs::create_dir_all(&path) {
-            return Err(AybError::Other {
-                message: format!("Unable to create entity path for {}: {}", entity_slug, e),
-            });
-        }
-    }
+    let path: PathBuf = [
```

Comment:
This pattern is great!!

Reply:
Aww thanks! I struggled for a while with it as a single file given the fact that there are journals, etc., and finally settled on every DB/iteration of a DB living in a directory :).